src/pch.hpp

```diff
@@ -172,3 +172,51 @@ struct fmt::formatter<E, std::enable_if_t<std::is_enum_v<E>, char>> : formatter<
 #include <eventpp/eventdispatcher.h>
 
 #include "lua/global/shared_object.hpp"
+
+#ifdef FEATURE_METRICS
+	#define METRICS_METHOD_LATENCY_MEASURE() metrics::method_latency measure(__METHOD_NAME__)
```

Comment:
I would rather have me method inside `metrics` do the branching. Macros are harder to maintain and nothing will protect us from calling `metrics::...` directly.